### ISSUE TITLE
Regression: `Workload ID wasm-tools is not recognized.`

### ISSUE THREAD

**Issue by radical** [dotnet/sdk#26967]
With installer version `7.0.100-rc.1.22403.4`, installing the `wasm-tools` workload fails with:

```
 $ ./dotnet workload install wasm-tools -v diag
Workload ID wasm-tools is not recognized.
```

IIUC, this corresponds to https://github.com/dotnet/installer/commit/6996a553 .
The previous version - `7.0.100-rc.1.22403.4` ( https://github.com/dotnet/installer/commit/a7f1b1c5) works fine.

There was only commit in from good to bad (https://github.com/dotnet/installer/compare/a7f1b1c5...6996a553) - `update dependencies from dotnet/sdk`). And the corresponding sdk side changes: https://github.com/dotnet/sdk/compare/80e6952...a121b7a .

It lists the workloads though:
```
 $ ./dotnet workload search

Workload ID            Description
---------------------------------------------------------------------------
android                .NET SDK Workload for building Android applications.
android-33             Preview support for Android API-33.
macos                  .NET SDK Workload for building macOS applications.
maui-android           .NET MAUI SDK for Android
maui-tizen             .NET MAUI SDK for Tizen
maui-windows           .NET MAUI SDK for Windows
runtimes-windows       workloads/runtimes-windows/description
wasm-experimental      workloads/wasm-experimental/description
wasm-tools             .NET WebAssembly build tools
```

This is breaking `runtime` CI also.

cc @lewing

**Comment by radical** [comment]
cc @steveisok I think this affects mobile workloads also.

```
 $ ./dotnet workload install macos
Workload ID macos is not recognized.
vscode@ [ /tmp/dbad ]
 $ ./dotnet workload install android
Workload ID android is not recognized.
```

**Comment by radical** [comment]
cc @gkulin @marcpopMSFT

**Comment by marcpopMSFT** [comment]
Repros with the zip install, doesn't repro in the dogfood environment. Probably caused by this codeflow but this isn't broken in 6.0.400: https://github.com/dotnet/sdk/pull/26796/files. Maybe it doesn't work with a change in main or something off in the merge?

**Comment by marcpopMSFT** [comment]
Daniel's fix to 6.0.400 should fix this so I merged the auto-flow to main: https://github.com/dotnet/sdk/pull/26971.  Should be fixed once that gets into installer.

**Comment by radical** [comment]
The flow PR has merged - https://github.com/dotnet/installer/pull/14258 . And the latest build is working. thank you!
